modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/shibari_worn.dm

```diff
@@ -0,0 +1,268 @@
+
+/obj/item/clothing/under/shibari/full
+	name = "Shibari fullbody ropes"
+	desc = "Bondage ropes that covers whole body"
+	icon_state = "shibari_fullbody"
+	rope_amount = 2
+
+//processing stuff
+/obj/item/clothing/under/shibari/full/process(delta_time)
+	var/mob/living/carbon/human/hooman = loc
+	if(tight == "low" && hooman.arousal < 20)
+		hooman.adjustArousal(0.6 * delta_time)
+		hooman.adjustPleasure(0.6 * delta_time)
+	if(tight == "medium" && hooman.arousal < 40)
+		hooman.adjustArousal(0.6 * delta_time)
+		hooman.adjustPleasure(0.6 * delta_time)
+	if(tight == "hard" && hooman.arousal < 70)
+		hooman.adjustArousal(0.6 * delta_time)
+		hooman.adjustPleasure(0.6 * delta_time)
+		if(hooman.pain < 40)
+			hooman.adjustPain(0.6 * delta_time)
+		if(prob(10))
+			hooman.adjustOxyLoss(5)
+
+/obj/item/clothing/gloves/shibari_hands
+	name = "Shibari arms bondage"
+	desc = "Bondage ropes that cover arms."
+	icon_state = "shibari_arms"
+	icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_clothing/lewd_gloves.dmi'
+	worn_icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/mob/lewd_clothing/lewd_gloves.dmi'
+	body_parts_covered = NONE
+	strip_delay = 100
+	breakouttime = 100
+	item_flags = DROPDEL
+	///color
+	var/current_color = "pink"
+	///tightness of the ropes can be low, medium and hard.
+	var/tight = "low"
+
+/obj/item/clothing/gloves/shibari_hands/Destroy()
+	var/obj/item/stack/shibari_rope/ropes = new(get_turf(src))
+	ropes.current_color = current_color
+	ropes.update_icon_state()
+	ropes.update_icon()
+	. = ..()
+
+//customization stuff
+/obj/item/clothing/gloves/shibari_hands/update_icon_state()
+	. = ..()
+	icon_state = "[initial(icon_state)]_[current_color]"
+	worn_icon_state = "[initial(icon_state)]_[current_color]"
+
+/obj/item/clothing/gloves/shibari_hands/ComponentInitialize()
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob)
+
+/obj/item/clothing/gloves/shibari_hands/Initialize()
+	. = ..()
+	update_appearance()
+
+//unequip stuff for adding rope to hands
+/obj/item/clothing/gloves/shibari_hands/attack_hand(mob/user)
+	. = ..()
+	if(.)
+		return
+	if(iscarbon(user))
+		var/mob/living/carbon/human/hooman = user
+		if(src == hooman.gloves)
+			if(do_after(hooman, HAS_TRAIT(hooman, TRAIT_RIGGER) ? 2 SECONDS : 10 SECONDS, target = src))
+				qdel(src)
+		else
+			return
+
+//stuff to apply mood event for perverts
+/obj/item/clothing/gloves/shibari_hands/equipped(mob/user, slot)
+	.=..()
+	var/mob/living/carbon/human/hooman = user
+	if(HAS_TRAIT(hooman, TRAIT_ROPEBUNNY))
+		hooman.apply_status_effect(/datum/status_effect/ropebunny)
+
+//same stuff as above but for dropping item
+/obj/item/clothing/gloves/shibari_hands/dropped(mob/user, slot)
+	.=..()
+	var/mob/living/carbon/human/hooman = user
+	if(HAS_TRAIT(hooman, TRAIT_ROPEBUNNY))
+		hooman.remove_status_effect(/datum/status_effect/ropebunny)
+
+/obj/item/clothing/shoes/shibari_legs
+	name = "Shibari arms bondage"
```

Comment:
Refer to the naming convention for items, aka, no capitalization for improper names.